lib/ansible/plugins/connection/ssh.py

```diff
@@ -253,6 +253,8 @@ def _send_initial_data(self, fh, in_data):
 
         try:
             fh.write(in_data)
+            if tty:
+                fh.write("__EOF__\n")
```

Comment:
I mentioned this on IRC, but for archival purposes I'll repeat myself here:
```python
#!/usr/bin/python

... python code ...

SOME_SHELL_SCRIPT = '''
#!/bin/sh
...

cat << __EOF__
hello
world
__EOF__

... more shell code ...
'''

... more python code...
```

is valid Python code.